apps/managedidentity/managedidentity.go

```diff
@@ -178,8 +209,24 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
+func retry(attempts int, delay time.Duration, c ops.HTTPClient, req *http.Request) (*http.Response, error) {
+	resp, err := c.Do(req)
+	if err == nil && !slices.Contains(retryStatusCodes, resp.StatusCode) {
```

Comment:
The `slices` package requires Go 1.21 but Azure SDK and MSAL still support 1.18. I don't think it's worth upgrading our min version for this because the alternative to `Contains` is a basic `for` loop